### ISSUE TITLE
internal/xkeyboard locale blinking on XF86AudioPlay keydown, becomes unclickable

### ISSUE THREAD

**Issue by kabalage** [polybar/polybar#1200]
Pressing _XF86AudioPlay_ key (or simulating it with `xdotool key XF86AudioPlay`) causes the locale to seemingly change between my configured keyboard layouts. After this the keyboard layout remains the same as before (altough the displayed indicator may say otherwise) and the locale becomes unclickable. Pressing caps lock fixes the module until I press _XF86AudioPlay_ again.

When the click does not work polybar logs:

```
warn: No matching input area found (btn=1)
```

Capturing the key _XF86AudioPlay_ in `.xbindkeysrc` or not does not make a difference.
I'm on Arch, using i3wm, installed polybar from [AUR](https://aur.archlinux.org/packages/polybar/), it's updated to version: `polybar 3.1.0-git`.

polybar config:

```
[bar/test]
monitor = eDP-1
modules-right = testxkeyboard

[module/testxkeyboard]
type = internal/xkeyboard
```

**Comment by patrick96** [comment]
Can you run polybar with `-s` then run `xdotool key XF86AudioPlay` and then post the whole output here.

Does clicking the xkeyboard module work before you press `XF86AudioPlay`?
Please run `setxkbmap -print` before and after pressing `XF86AudioPlay` and also post the output.

This whole thing doesn't really look like it's polybar's fault, polybar only reads the keyboard layouts and provides a click handler, if you have multiple layouts defined. What I suspect is happening, is that whatever you are doing, makes you somehow only have a single layout, and thus polybar doesn't have a click handler anymore.

**Comment by kabalage** [comment]
I removed my .xbindkeysrc, entered `killall xbindkeys`, rebooted, but `xdotool key XF86AudioPlay` always caused bugs.

It turns out every xdotool key command causes this glitch, not just XF86AudioPlay.
Pressing xkeyboard works before xdotool commands and after caps lock. Between the two events, it shows `us` and it's not clickable. Clicking another clickable module does not fix the issue.

Here is the polybar output: (added line numbers, comments for lines are below the output)

```
1: %{r}%{A1:xkeyboard/switch:}hu%{A}
2: %{r}us
3: %{r}%{A1:xkeyboard/switch:}hu%{A}
4: %{r}us
5: %{r}%{A1:xkeyboard/switch:}hu%{A}
6: %{r}%{A1:xkeyboard/switch:}hu Caps Lock%{A}
7: %{r}%{A1:xkeyboard/switch:}hu%{A}
```

1. polybar started.
2. executed `xdotool key XF86AudioPlay`
3. appeared right after 2. (altough it says `hu`, when running without `-s` it stays on `us`, this row is seemingly not rendered)
4. executed `xdotool key XF86AudioPlay`
5. appeared right after 4.
6. pressed caps lock
7. pressed caps lock again

`setxkbmap -print` output:

```
xkb_keymap {
	xkb_keycodes  { include "evdev+aliases(qwertz)"	};
	xkb_types     { include "complete"	};
	xkb_compat    { include "complete"	};
	xkb_symbols   { include "pc+hu+us:2+inet(evdev)"	};
	xkb_geometry  { include "pc(pc105)"	};
};
```

Maybe I configured my keyboard wrong.
Maybe xdotool messes with the selected layout and it's doing it too fast and the second event does not cause a render?

**Comment by patrick96** [comment]
Maybe there is more to this, than I first thought. aeSee5eiqu on IRC had a similar issue. And they said the following:

> < aeSee5eiqu> [01:05:04] patrick96: thanks! I figured it out: systray icons have been loading in .xinitrc BEFORE running sxhkd, messing up polybar.

I'm not sure what the systray icons have to do with it though.
Do you use any software to configure your keyboard that might be messing this up?

**Comment by kabalage** [comment]
- Disabling tray in polybar and killing apps using it did not help.
- I only use xbindkeys, and it starts before polybar. (but it does not matter if I restart it later)
- Besides xbindkeys i3 also has keybindings but I only configured window management stuff there, nothing references XF86AudioPlay or the key I bound it to (pause/break).
- I just tried a keyboard with a physical button for XF86AudioPlay and the bar does not flicker so definitely xdotool does something weird that is not handled by polybar.
- I've read the xdotool manual, tried different things with the same result.

These are my relevant bindings in `.xbindkeysrc`:

```
"playerctl play-pause"
  XF86AudioPlay

"xdotool key XF86AudioPlay"
  Pause + Release
```
